src/cryptonote_core/service_node_list.cpp

```diff
@@ -3918,6 +4203,27 @@ crypto::public_key service_node_list::get_random_pubkey() {
     }
 }
 
+// Function to get an iterator for the first service node info
+service_nodes::service_nodes_infos_t::iterator service_node_list::get_first_pubkey_iterator() {
+    std::lock_guard lock{m_sn_mutex};
```

Comment:
Sorted in https://github.com/oxen-io/oxen-core/commit/5c2bebbd276140824dafc05553c90a757c30b7bb

I removed the API because I've achieved what we needed to do more optimally without needing to expose iterators. We can re-add in the future if a use-case requires.